From 6e5418c8c11d7b59c25f1a6dd5c2c532e4862ad7 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Tue, 21 Jul 2020 08:28:25 +0700
Subject: [PATCH 204/504] ARM: dts iwg21m: enable TPU

This patch enables TPU on iwg21m, TPU is used for buzzer, fan.
This commit enables all of them.

Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 1e5dea7..44d0bbe 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -42,6 +42,24 @@
 		#size-cells = <1>;
 	};
 
+	buzzer {
+		compatible = "pwm-backlight";
+		pwms = <&tpu 0 5000000 GPIO_ACTIVE_HIGH>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <7>;
+		pinctrl-0 = <&buzzer_pins>;
+		pinctrl-names = "default";
+	};
+
+	fan {
+		compatible = "pwm-backlight";
+		pwms = <&tpu 1 5000000 GPIO_ACTIVE_HIGH>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <7>;
+		pinctrl-0 = <&fan_pins>;
+		pinctrl-names = "default";
+	};
+
 	reg_0p0v: 0p0v {
 		compatible = "regulator-fixed";
 		regulator-name = "0P0V";
@@ -194,6 +212,11 @@
 		function = "avb";
 	};
 
+	buzzer_pins: buzzer {
+		groups = "tpu0_to0";
+		function = "tpu0";
+	};
+
 	can0_pins: can0 {
 		groups = "can0_data_d";
 		function = "can0";
@@ -209,6 +232,11 @@
 		function = "eth";
 	};
 
+	fan_pins: fan {
+		groups = "tpu0_to1";
+		function = "tpu0";
+	};
+
 	mmc1_8_pins: mmc1_8 {
 		groups = "mmc1_data8", "mmc1_ctrl";
 		function = "mmc1";
@@ -347,3 +375,7 @@
 &ssi4 {
 	shared-pin;
 };
+
+&tpu {
+	status = "okay";
+};
-- 
2.7.4

